scss/mixins/_pagination.scss

```diff
@@ -11,17 +11,68 @@
 
 // scss-docs-start pagination-max-items-mixin
 // Boosted mod
-// Ensure we only display `n` items:
-// - the first `n/2` items
-// - the last `n/2` items
-// - including the .active one (so maybe not the `n/2`th last item)
+// The pagination list ((m) items) should display exactly (n) items and is built in 3 parts:
+// - the begining of the list is made of the first (n/2) items
+// - the end of the list is made of the last (n/2) items (or (n/2 - 1) depending on the active element)
+// - the last part of the list is made of the active element
+// - n should be greater than 7 since we want at least two arrows, the first, the active and the last and might have 2 three dots cases for a11y and design reasons
 @mixin pagination-max-items($pagination-max-items) {
-  display: none;
+  @if $enable-responsive-pagination {
+    $pagination-max-items: max(7, $pagination-max-items);
+    $half-first-items: $pagination-max-items * .5;
+    $half-last-items: $pagination-max-items * .5;
+    @if $pagination-max-items % 2 != 0 {
+      $half-first-items: ($pagination-max-items - 1) * .5;
+      $half-last-items: ($pagination-max-items + 1) * .5;
+    }
+    display: none;
 
-  &.active,
-  &:nth-child(-n+#{$pagination-max-items * .5}),
-  &:nth-last-child(-n+#{$pagination-max-items * .5}) {
-    display: list-item;
+    // rule 0: don't display any item unless they are part of the following list:
+    //  - the active item
+    //  - the (n/2 - 1) first items
+    //  - the (n/2) last items
+    //  - the (n/2) first item where m equals n
+    &.active,
+    &:nth-child(-n + #{$half-first-items - 1}),
+    &:nth-last-child(-n + #{$half-last-items}),
+    &:nth-child(#{$half-first-items}):nth-last-child(#{$half-last-items + 1}) {
+      display: flex;
+    }
+
+    // rule 1: display `...` on:
+    //  - the (n/2 - 1) first item
+    //  - the active item when it's:
+    //      - not in (n/2 - 1) first items
+    //      - not in (n/2) last items
+    //      - not the (n/2) first item when m equals n
+    &:nth-child(#{$half-first-items - 1}),
+    &:nth-child(#{$half-first-items - 1}) ~ .active:not(:nth-last-child(-n + #{$half-last-items})):not(:nth-child(#{$half-first-items}):nth-last-child(#{$half-last-items + 1})) {
```

Comment:
Two cases to investigate:

![screenshot-deploy-preview-1370--boosted netlify app-2022 12 20-10_12_20](https://user-images.githubusercontent.com/92363071/208630415-6a6fed9b-e505-40fd-9f1d-84892033ac09.png)
![screenshot-deploy-preview-1370--boosted netlify app-2022 12 20-10_14_43](https://user-images.githubusercontent.com/92363071/208630432-862a2ee0-4c1c-47ad-800e-7f475ad09c8f.png)